### ISSUE TITLE
[feature request] Sparse output option for topk, min, max

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#34951]
I wonder if PyTorch hybrid sparse/dense tensors can be used to represent outputs of dimension-wise topk / min / max. Unfortunately I couldn't completely understand from docs if they can be used for that (my current understanding is that they could not, hence feature request).

Probably pruning work is related as well.

My current impl storing directly indices/values from topk: https://gist.github.com/vadimkantorov/c473a181017e654444cbdfbcafa265e2

Usecase: store logits/log_probs compactly on disk
Context: https://discuss.pytorch.org/t/sparse-torch-topk-can-hybrid-sparse-dense-tensors-help/71832/2

**Comment by colesbury** [comment]
I'll let @vincentqb comment, but it seems like a sparse tensor constructor that accepts the output format of the indices + values of topk/min/max would be sufficient without having to complicate the topk/min/max API.

**Comment by vadimkantorov** [comment]
@colesbury Sparse tensors waste a lot of memory on storing index tuples for all remaining dimensions (the index pattern is very regular, since reduction is typically done only across one dimension). So I was thinking about hybrid sparse/dense tensors.

**Comment by vadimkantorov** [comment]
I guess, this might also be useful for some pruning methods as well (those that keep topk largest weights). Of course more useful

**Comment by collinmccarthy** [comment]
This feature would be awesome for many different sparsity-based algorithms. Thanks for your work on this!
